code/game/machinery/vending.dm

```diff
@@ -221,7 +225,7 @@
 		var/price = prices[typepath]
 		if(isnull(amount)) amount = 1
 
-		var/atom/temp = new typepath(null)
+		var/obj/item/temp = new typepath(null)

```

Comment:
There's no need to do a new here.
You can just cast the typepath and use initial() for the vars.